index.js

```diff
@@ -158,6 +185,38 @@ function generatePowerGeneration (geo, subIndicators) {
   }
 }
 
+// Generate data for the Clean Energy Investment Chart for a single country
+function generateEnergyInvestment (geo, investments) {
+  return {
+    'id': 'cleanEnergyInvestment',
+    'meta': {
+      'label-y': 'usd/m',
+      'label-x': 'year',
+      'title': 'Clean Energy Investment'
+    },
+    'data': investments
+      .filter(i => i.geography === geo.name)
+      .reduce((a, b) => {
+        let v = {
+          value: b.value,
+          year: b.year
+        }
+
+        let match = a.find(o => o.name === b.sector)
+
+        if (!match) {
+          a.push({
+            name: b.sector,
+            values: [v]
+          })
+        } else {
+          match['values'].push(v)
```

Comment:
There's a lot here being modified by reference making it hard to track. If you have the time to change it please avoid this approach.